src/com/amazon/corretto/crypto/provider/EvpSignatureBase.java

```diff
@@ -113,13 +122,81 @@ protected void engineSetParameter(final String param, final Object value) throws
     @Override
     protected synchronized void engineSetParameter(final AlgorithmParameterSpec params)
             throws InvalidAlgorithmParameterException {
-        if (params != null) {
-            throw new InvalidAlgorithmParameterException("No parameters supported by this algorithm");
+        if (params instanceof PSSParameterSpec) {
+            if (!isBufferEmpty()) {
+                throw new IllegalStateException("Cannot update PSS parameters with buffered data, reset Signature.");
+            }
+            final PSSParameterSpec pssParams = (PSSParameterSpec) params;
+            if (keyType_ != EvpKeyType.RSA || paddingType_ != RSA_PKCS1_PSS_PADDING) {
+                throw new InvalidAlgorithmParameterException("PSS params only supported for RSASSA-PSS signatures");
+            }
```

Comment:
Move this check before checking for an empty buffer. This is the more important failure.

Reply:
done.